Integrations/integration-McAfeeWebGateway.yml

```diff
@@ -0,0 +1,328 @@
+description: Blacklist/Whitelist handling
+configuration:
+- display: 'URL (e.g. https://192.168.100.55:12345) '
+  name: url
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: ""
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Use system proxy settings
+  name: useproxy
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |-
+    // handle '/' at the end of the url
+    var base_url = params.url.slice(0, params.url.length - params.url.match('/*$')[0].length) + '/Konfigurator/REST';
+    var session_id = '';
+
+    //body should be in xml format
+    function sendRequest(method, url_suffix, headers, body) {
+        headers = headers || {};
+        body = body || '';
+
+        // add default headers
+        if (!("Accept" in headers)) {
+            headers.Accept = ['application/mwg+xml'];
+        }
+        if (!("Content-Type" in headers)) {
+            headers['Content-Type'] = ['application/mwg+xml'];
+        }
+        headers.Cookie = ['JSESSIONID=' + session_id];
+
+        var res = http(
+            base_url + url_suffix,
+            {
+                Method: method,
+                Headers: headers,
+                Body: body,
+            },
+            params.insecure,
+            params.useproxy
+        );
+
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+        return res.Body;
+    }
+
+    function login() {
+        var cmd_url = '/login?userName=' + params.credentials.identifier + '&pass=' + params.credentials.password;
```

Comment:
probably better to use url encode of a struct with userName: params.credentials.identifier, and pass: params.credentials.password, and it will be more clear than this long text